addr.go

```diff
@@ -46,3 +48,21 @@ func addrEqual(a, b net.Addr) bool {
 
 	return aType == bType && aIP.Equal(bIP) && aPort == bPort
 }
+
+// AddrPort is  an IP and a port number.
+type AddrPort [18]byte
+
+func toAddrPort(addr net.Addr) AddrPort {
+	var ap AddrPort
+	switch addr := addr.(type) {
+	case *net.UDPAddr:
```

Comment:
Is it possible to combine the two case branches?
E.g.;

```go
switch addr := addr.(type) {
	case *net.UDPAddr, *net.TCPAddr:
	    [...]
}
```

Reply:
It seems impossible to typecast two different structures with switch case

Reply:
Ah okay thanks for checking :)